x/leverage/keeper/grpc_query.go

```diff
@@ -269,3 +269,37 @@ func (q Querier) BadDebts(
 
 	return &types.QueryBadDebtsResponse{Targets: targets}, nil
 }
+
+func (q Querier) MaxWithdraw(
+	goCtx context.Context,
+	req *types.QueryMaxWithdraw,
+) (*types.QueryMaxWithdrawResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+	if req.Address == "" {
+		return nil, status.Error(codes.InvalidArgument, "empty address")
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	addr, err := sdk.AccAddressFromBech32(req.Address)
+	if err != nil {
+		return nil, err
+	}
+
+	uToken, err := q.Keeper.maxWithdraw(ctx, addr, req.Denom)
+	if err != nil {
+		return nil, err
+	}
+
+	token, err := q.Keeper.ExchangeUToken(ctx, uToken)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Should be fairly straightforward, but to clarify:
- the `uToken` amount is what you can withdraw, in uTokens
- the `token` amount converts it to base tokens as a courtesy - probably best for frontend when displaying withdraw slider (this is the token value at 100%)